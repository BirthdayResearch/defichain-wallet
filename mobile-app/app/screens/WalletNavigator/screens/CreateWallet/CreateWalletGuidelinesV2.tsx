import { Feather } from '@expo/vector-icons'
import { StackScreenProps } from '@react-navigation/stack'
import { useState } from 'react'
import * as React from 'react'
import { View, Image } from 'react-native'
import Checkbox from 'expo-checkbox'
import { ThemedIcon, ThemedScrollViewV2, ThemedTextV2, ThemedTouchableOpacityV2, ThemedViewV2 } from '@components/themed'
import { getColor, tailwind } from '@tailwind'
import { translate } from '@translations'
import { WalletParamListV2 } from '../../WalletNavigator'
import DarkNewWallet from '@assets/images/darkNewWallet.png'
import LightNewWallet from '@assets/images/lightNewWallet.png'
import { useThemeContext } from '@shared-contexts/ThemeProvider'
import { ButtonV2 } from '@components/ButtonV2'

type Props = StackScreenProps<WalletParamListV2, 'CreateWalletGuidelines'>

interface GuidelineItem {
  icon: React.ComponentProps<typeof Feather>['name']
  title: string
}

const guidelines: GuidelineItem[] = [
  {
    title: 'Write the words on paper. Take note of their correct spelling and order.',
    icon: 'edit-2'
  },
  {
    title: 'Secure them in a safe place. Store them offline at a place only you have access. Keep them private and do not share it with anyone.',
    icon: 'lock'
  }
]

export function CreateWalletGuidelinesV2 ({ navigation }: Props): JSX.Element {
  const [isEnabled, setIsEnabled] = useState(false)
  const toggleSwitch = (): void => setIsEnabled(previousState => !previousState)
  const { isLight } = useThemeContext()

  return (
    <ThemedScrollViewV2
      contentContainerStyle={tailwind('pt-12 px-5 pb-16')}
      style={tailwind('flex-1')}
    >
      <View style={tailwind('flex flex-row justify-center')}>
        <Image
          style={{ width: 200, height: 136 }}
          source={isLight ? LightNewWallet : DarkNewWallet}
        />
      </View>
      <ThemedTextV2
        style={tailwind('mt-7 px-3 text-base font-normal-v2 text-center')}
      >
        {translate('screens/Guidelines', 'You will be shown 24 recovery words on the next screen. Keep your 24-word recovery safe as it will allow you to recover access to the wallet')}
      </ThemedTextV2>
      <ThemedTouchableOpacityV2
        style={tailwind('mt-2 flex-row items-center justify-center')}
        light={tailwind('border-b-0')}
        dark={tailwind('border-b-0')}
        onPress={() => navigation.navigate('RecoveryWordsFaq')}
        testID='recovery_words_faq'
      >
        <ThemedIcon
          light={tailwind('text-mono-light-v2-900')}
          dark={tailwind('text-mono-dark-v2-900')}
          iconType='MaterialIcons'
          name='help'
          size={16}
        />
        <ThemedTextV2
          style={tailwind('pl-1.5 text-sm text-center font-semibold-v2')}
        >
          {translate('screens/Guidelines', 'Learn more')}
        </ThemedTextV2>
      </ThemedTouchableOpacityV2>
      <View style={tailwind('px-6 mt-12')}>
        {guidelines.map((g, i) => (
          <View
            key={i}
            style={tailwind('flex flex-row items-start my-3')}
          >
            <ThemedViewV2
              light={tailwind('bg-mono-light-v2-700')}
              dark={tailwind('bg-mono-dark-v2-700')}
              style={tailwind('flex h-6 w-6 flex-row items-center justify-center rounded-full mt-1')}
            >
              <ThemedIcon
                light={tailwind('text-mono-light-v2-100')}
                dark={tailwind('text-mono-dark-v2-100')}
                iconType='Feather'
                name={g.icon}
                size={14}
              />
            </ThemedViewV2>
            <ThemedTextV2
              light={tailwind('text-mono-light-v2-700')}
              dark={tailwind('text-mono-dark-v2-700')}
              style={tailwind('flex-1 text-xs ml-4 font-normal-v2')}
            >
              {translate('screens/Guidelines', g.title)}
            </ThemedTextV2>
          </View>
        ))}
        <View style={tailwind('mt-3 flex flex-row items-start')}>
          <Checkbox
            value={isEnabled}
            onValueChange={toggleSwitch}
            style={tailwind('h-6 w-6 mt-1 rounded')}
            color={isEnabled ? getColor('brand-v2-500') : undefined}
          />
          <ThemedTouchableOpacityV2
            light={tailwind('border-b-0')}
            dark={tailwind('border-b-0')}
            style={tailwind('flex-1')}
            onPress={toggleSwitch}
          >
            <ThemedTextV2
              light={tailwind('text-mono-light-v2-700')}
              dark={tailwind('text-mono-dark-v2-700')}
              style={tailwind('text-xs ml-4 font-normal-v2')}
            >
              {translate('screens/Guidelines', 'I understand it is my responsibility to keep my recovery words secure. Losing them will result in the irrecoverable loss of access to my wallet funds.')}
            </ThemedTextV2>
          </ThemedTouchableOpacityV2>
        </View>
      </View>
      <ButtonV2
        disabled={!isEnabled}
        style={tailwind('rounded')}
        label={translate('screens/Guidelines', 'Create wallet')}
        styleProps='mt-12 mx-7'
        onPress={() => navigation.navigate('CreateMnemonicWallet')}
        testID='create_recovery_words_button'
      />
    </ThemedScrollViewV2>
  )
}
