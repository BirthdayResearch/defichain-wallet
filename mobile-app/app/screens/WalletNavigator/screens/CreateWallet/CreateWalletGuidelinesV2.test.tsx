import { render } from '@testing-library/react-native'
import { CreateWalletGuidelinesV2 } from './CreateWalletGuidelinesV2'

jest.mock('@shared-contexts/ThemeProvider')
describe('create wallet guidelines v2', () => {
  it('should match snapshot', () => {
    const navigation: any = {
      navigate: jest.fn()
    }
    const route: any = {}
    const rendered = render(
      <CreateWalletGuidelinesV2
        navigation={navigation}
        route={route}
      />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
