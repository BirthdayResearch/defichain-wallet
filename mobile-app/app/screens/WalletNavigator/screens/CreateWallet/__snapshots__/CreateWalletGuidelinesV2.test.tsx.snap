// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create wallet guidelines v2 should match snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 64,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 48,
    }
  }
  style={
    Array [
      Object {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      Object {
        "backgroundColor": "rgba(242, 242, 242, 1)",
      },
    ]
  }
>
  <View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 136,
            "width": 200,
          }
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontSize": 16,
              "lineHeight": 24,
              "marginTop": 28,
              "textAlign": "center",
            },
            Object {
              "color": "rgba(18, 18, 18, 1)",
            },
          ],
        ]
      }
    >
      You will be shown 24 recovery words on the next screen. Keep your 24-word recovery safe as it will allow you to recover access to the wallet
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontFamily": "SemiBoldFont",
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 24,
              "marginTop": 8,
              "textAlign": "center",
            },
            Object {
              "color": "rgba(18, 18, 18, 1)",
            },
          ],
        ]
      }
    >
      Learn more
    </Text>
    <View
      style={
        Object {
          "marginTop": 48,
          "paddingLeft": 12,
          "paddingRight": 12,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "marginBottom": 12,
            "marginTop": 12,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 9999,
                "borderBottomRightRadius": 9999,
                "borderTopLeftRadius": 9999,
                "borderTopRightRadius": 9999,
                "flexDirection": "row",
                "height": 24,
                "justifyContent": "center",
                "marginTop": 4,
                "width": 24,
              },
              Object {
                "backgroundColor": "rgba(43, 43, 43, 1)",
              },
            ]
          }
        >
          <Text />
        </View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontSize": 12,
                  "lineHeight": 16,
                  "marginLeft": 16,
                },
                Object {
                  "color": "rgba(43, 43, 43, 1)",
                },
              ],
            ]
          }
        >
          Write the words on paper. Take note of their correct spelling and order.
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "marginBottom": 12,
            "marginTop": 12,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 9999,
                "borderBottomRightRadius": 9999,
                "borderTopLeftRadius": 9999,
                "borderTopRightRadius": 9999,
                "flexDirection": "row",
                "height": 24,
                "justifyContent": "center",
                "marginTop": 4,
                "width": 24,
              },
              Object {
                "backgroundColor": "rgba(43, 43, 43, 1)",
              },
            ]
          }
        >
          <Text />
        </View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontSize": 12,
                  "lineHeight": 16,
                  "marginLeft": 16,
                },
                Object {
                  "color": "rgba(43, 43, 43, 1)",
                },
              ],
            ]
          }
        >
          Secure them in a safe place. Store them offline at a place only you have access. Keep them private and do not share it with anyone.
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "marginTop": 12,
          }
        }
      >
        <View
          accessibilityRole="checkbox"
          accessibilityState={
            Object {
              "checked": false,
              "disabled": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderColor": "#657786",
                "borderRadius": 2,
                "borderWidth": 2,
                "height": 20,
                "width": 20,
              },
              Object {
                "borderBottomLeftRadius": 4,
                "borderBottomRightRadius": 4,
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
                "height": 24,
                "marginTop": 4,
                "width": 24,
              },
              false,
              false,
              undefined,
              false,
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Array [
                Object {
                  "fontSize": 12,
                  "lineHeight": 16,
                  "marginLeft": 16,
                },
                Object {
                  "color": "rgba(43, 43, 43, 1)",
                },
              ],
            ]
          }
        >
          I understand it is my responsibility to keep my recovery words secure. Losing them will result in the irrecoverable loss of access to my wallet funds.
        </Text>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(166, 166, 166, 1)",
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": 16,
            "marginRight": 16,
            "marginTop": 48,
            "opacity": 1,
            "paddingBottom": 14,
            "paddingLeft": 14,
            "paddingRight": 14,
            "paddingTop": 14,
          }
        }
        testID="create_recovery_words_button"
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "RegularFont",
                "fontSize": 16,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Object {
                "color": "rgba(242, 242, 242, 1)",
                "fontFamily": "BoldFont",
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
        >
          Create wallet
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
